trace-test-application/src/androidTest/java/io/bitrise/trace/testapp/BaseUiTest.java

```diff
@@ -141,27 +152,66 @@ private String getLauncherPackageName() {
     }
 
     /**
-     * Check if there is an ANR dialog and closes the app if there is one.
+     * Takes and stores a screenshot of the device.
+     *
+     * @param testEvent a {@link TestEvent} value to postfix the tested method.
      */
-    private void closeAnrDialog() {
-        final UiObject2 anrDialog = uiDevice.wait(Until.findObject(By.text("Close app")), WAIT_FOR_IDLE_TIMEOUT);
-        if (anrDialog != null) {
-            anrDialog.click();
-        }
+    protected void takeScreenShot(@NonNull final TestEvent testEvent) {
+        final Bitmap screenshotBitmap = getInstrumentation().getUiAutomation().takeScreenshot();
+        final String screenShotFileName = getTestReportFileBaseName(testEvent);
+
+        storeScreenshot(screenshotBitmap, screenShotFileName);
     }
 
     /**
-     * Takes and stores a screenshot of the device.
+     * Gets the base name for collected test data files, which is the current time concatenated with the name of the
+     * test and the {@link TestEvent}.
      *
-     * @param event a {@link ScreenshotEvent} value to postfix the tested method.
+     * @param testEvent the current TestEvent.
+     * @return the base name for test data files.
      */
-    protected void takeScreenShot(@NonNull final ScreenshotEvent event) {
-        final Bitmap screenshotBitmap = getInstrumentation().getUiAutomation().takeScreenshot();
-        final SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyyMMdd_hh:mm:ss");
+    @NonNull
+    private String getTestReportFileBaseName(@NonNull final TestEvent testEvent) {
+        final SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyyMMdd_hhmmss");
         final String timeStamp = simpleDateFormat.format(Calendar.getInstance().getTime());
-        final String screenShotFileName = timeStamp + "_" + testName.getMethodName() + "_" + event;
+        return timeStamp + "_" + testName.getMethodName() + "_" + testEvent;
+    }
 
-        storeScreenshot(screenshotBitmap, screenShotFileName);
+    /**
+     * Tries to press the button to wait for the app that is not responding.
+     *
+     * @param anrDialog the given ANR dialog.
+     * @return {@code true} if there was an ANR dialog, {@code false} otherwise.
+     */
+    private static boolean clickWaitForAnrDialog(@NonNull final UiObject anrDialog) {
```

Comment:
nit: i'm not sure `clickWaitForAnrDialog` is descriptive - perhaps `checkForAnrDialogAndAction`? as it's first checking if one actually exists, and only doing something if it does?

Reply:
Awesome. that's a great idea :) Thank you!